lib/VM/JSLib/Intl.cpp

```diff
@@ -386,6 +387,33 @@ ExecutionStatus checkOptions(Runtime &runtime, const platform_intl::Options &) {
   return ExecutionStatus::RETURNED;
 }
 
+template <>
+ExecutionStatus checkOptions<platform_intl::DateTimeFormat>(
+    Runtime &runtime,
+    const platform_intl::Options &options) {
+  // https://www.ecma-international.org/wp-content/uploads/ECMA-402_9th_edition_june_2022.pdf
+  //  InitializeDateTimeFormat # 42
+  const bool hasExplicitFormatComponents = options.find(u"weekday") != options.end() ||
+      options.find(u"era") != options.end() ||
+      options.find(u"year") != options.end() ||
+      options.find(u"month") != options.end() ||
+      options.find(u"day") != options.end() ||
+      options.find(u"day") != options.end() ||
+      options.find(u"dayPeriod") != options.end() ||
+      options.find(u"hour") != options.end() ||
+      options.find(u"minute") != options.end() ||
+      options.find(u"second") != options.end() ||
+      options.find(u"fractionalSecondDigits") != options.end() ||
+      options.find(u"timeZoneName") != options.end();
+  const bool hasStyle = options.find(u"dateStyle") != options.end() || options.find(u"timeStyle") != options.end();
+  if ( hasStyle && hasExplicitFormatComponents) {
```

Comment:
Add a comment

```
// 42. If dateStyle is not undefined or timeStyle is not undefined, then
//    a. If hasExplicitFormatComponents is true, then
//        i. Throw a TypeError exception.
```

before the if to link the behavior with the Spec section defining it.